packages/components/src/components/scroller/scroller.tsx

```diff
@@ -111,18 +112,24 @@ export class Scroller {
     }
   }
 
+  public componentShouldUpdate(
+    newVal: unknown,
+    oldVal: unknown,
+    propName: keyof InstanceType<typeof Scroller>
+  ): boolean {
+    return (
+      !isDeepEqual(newVal, oldVal) &&
+      !(propName === 'scrollToPosition' && !isScrollable(this.isNextHidden, this.isPrevHidden)) // should only update if scrollable
```

Comment:
Switching conditions around should be a little bit more performant.
In terms of lifecycle hook order it was better before since `componentDidLoad` happens first?